### PR TITLE
make the agent deploy controller watch the managed cluster

### PR DESCRIPTION
**Pull request by zhujian7**
Provide an option for developer to make the agent deploy controller watches the managed cluster, so if there are any changes on the managed cluster, the agent deploy controller can reconcile quickly

for example: if a helm addon wants managed cluster annotation changing trigger the agent redeploy:
```
.WithAgentDeployTriggerClusterFilter(func(old, new *clusterv1.ManagedCluster) bool {
			return !equality.Semantic.DeepEqual(old.Annotations, new.Annotations)
		})
```

issue reference: https://github.com/open-cluster-management-io/addon-framework/issues/198

### CODE REVIEW

File: pkg/agent/inteface.go [hunk 1]
```diff
@@ -76,6 +76,19 @@ type AgentAddonOptions struct {
...
+	// field to set what information they need, otherwise the expected/up-to-date agent may be deployed delayed since
+	// the default filter func returns false when the ManagedCluster resource is updated.
+	//
+	// For example, the agentAddon needs information from the ManagedCluster annotation, it can set the filter function
+	// like:
+	//
+	//	AgentDeployClusterInformerFilter: func(old, new *clusterv1.ManagedCluster) bool {
+	//	 return !equality.Semantic.DeepEqual(old.Annotations, new.Annotations)
+	//	}
+	AgentDeployClusterInformerFilter func(old, new *clusterv1.ManagedCluster) bool
```
Comment (qiujian16) [inline pkg/agent/inteface.go]:
```suggestion
	AgentDeployTriggerClusterFilter func(old, new *clusterv1.ManagedCluster) bool
```
something like this?

File: pkg/agent/inteface.go [hunk 2]
```diff
@@ -76,6 +76,19 @@ type AgentAddonOptions struct {
 	// SupportedConfigGVRs is a list of addon supported configuration GroupVersionResource
 	// each configuration GroupVersionResource should be unique
 	SupportedConfigGVRs []schema.GroupVersionResource
+
+	// AgentDeployClusterInformerFilter defines the filter function for the cluster informer when deploying the addon
```
Comment (qiujian16) [inline pkg/agent/inteface.go]: the filter func to trigger the agent deploy/redploy when cluster info is changed.

Comment (zhujian7) [reply pkg/agent/inteface.go]: fixed

File: pkg/agent/inteface.go (hunk 1 above)
Comment (zhujian7) [reply pkg/agent/inteface.go]: ok, fixed

**Pull request merged**